src/bot.py

```diff
@@ -151,15 +141,13 @@ def main(self):
 
 if __name__ == '__main__':
     try:
-        FLAG = True
         TwitterBot = Bot()
         TwitterBot.main()
-        
-    except Exception as ex:
-        exception = type(ex).__name__
-        
-        if exception == 'TweepError':
-            logging.critical("Authentication Error!")
-            logging.info("Please validate your credentials.")
-            time.sleep(10)
-            quit()
\ No newline at end of file
+    except tweepy.RateLimitError:
+        logging.critical("Tweeting failed due to ratelimit. Waiting {} more minutes.".format(cooldown))
+        time.sleep(cooldown)
+    except tweepy.TweepError:
+      logging.critical("Authentication Error!")
+      logging.info("Please validate your credentials.")
+      time.sleep(10)

```

Comment:
This is starting to look WAY better, although I'd remove `time.sleep()`

Reply:
Noted